### PR TITLE
tidb-functions: update tidb-functions.md doc

### PR DESCRIPTION
**Pull request by TonsnakeLin**
Signed-off-by: TonsnakeLin <lpbgytong@163.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?
https://github.com/pingcap/tidb/pull/30659

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TonsnakeLin** [comment]
cc @MyonKeminta  @youjiali1995 @cfzjywxk

File: functions-and-operators/tidb-functions.md [hunk 1]
```diff
@@ -329,3 +329,42 @@ select tidb_decode_sql_digests(@digests, 10);
...
+TIDBShardExpr ::=
+    "TIDB_SHARD" "(" expr ")"
+```
+
+### 示例
+
+{{< copyable "sql" >}}
+
+```sql
+select TIDB_SHARD(12373743746);
```
Comment (cfzjywxk) [inline functions-and-operators/tidb-functions.md:353]: Do we need to add some query examples here to explain how to use it reading.

File: functions-and-operators/tidb-functions.md [hunk 2]
```diff
@@ -329,3 +329,42 @@ select tidb_decode_sql_digests(@digests, 10);
 
 - [`Statement Summary Tables`](/statement-summary-tables.md)
 - [`INFORMATION_SCHEMA.TIDB_TRX`](/information-schema/information-schema-tidb-trx.md)
+
+## TIDB_SHARD
+
+TIDB_SHARD 函数用于创建一个 SHARD INDEX 来打散热点索引，如下所示。二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在单调递增或递减而导致的热点，前缀 `tidb_shard(a)` 将热点打散，提升集群扩展性。
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:335]:
```suggestion
`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。当二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在因单调递增或递减而导致的热点时，索引的前缀 `tidb_shard(a)` 将热点打散，从而提升集群扩展性。
```

File: functions-and-operators/tidb-functions.md [hunk 3]
```diff
@@ -329,3 +329,42 @@ select tidb_decode_sql_digests(@digests, 10);
...
+
+### 语法图
+
+```
+TIDBShardExpr ::=
+    "TIDB_SHARD" "(" expr ")"
+```
+
+### 示例
+
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:349]:
```suggestion

本示例会使用 `TIDB_SHARD` 函数计算 `12373743746` 的 SHARD 值。

```

File: functions-and-operators/tidb-functions.md [hunk 4]
```diff
@@ -329,3 +329,42 @@ select tidb_decode_sql_digests(@digests, 10);
...
+| TIDB_SHARD(12373743746) |
++-------------------------+
+|                     184 |
++-------------------------+
+1 row in set (0.00 sec)
+```
+
+以上示例使用 `TIDB_SHARD` 计算 `12373743746` 的 SHARD 值。
+
+### MySQL 兼容性
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:367]:
```suggestion

### MySQL 兼容性
```

File: functions-and-operators/tidb-functions.md (hunk 2 above)
Comment (en-jin19) [reply functions-and-operators/tidb-functions.md:335]: @TonsnakeLin 补充了一些 release notes 中的内容，可以请看下是否合适吗？如果合适的话，可以请帮忙在文档内补充如何进行 341 行中所说的验证吗？

File: functions-and-operators/tidb-functions.md [hunk 5]
```diff
@@ -329,3 +329,49 @@ select tidb_decode_sql_digests(@digests, 10);
...
+
+`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。当二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在因单调递增或递减而导致的热点时，索引的前缀 `tidb_shard(a)` 将热点打散，从而提升集群扩展性。
+
+在高吞吐写入、点查询、批量点查询场景下，你可以在保留原有的查询条件的情况下使用该函数，以提升写入性能。
+
+> **注意：**
+>
+> 如果业务中使用范围查询打散后的数据，可能造成性能回退，请验证后使用。
+
+
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:343]:
```suggestion

```

File: functions-and-operators/tidb-functions.md [hunk 6]
```diff
@@ -329,3 +329,49 @@ select tidb_decode_sql_digests(@digests, 10);
 
 - [`Statement Summary Tables`](/statement-summary-tables.md)
 - [`INFORMATION_SCHEMA.TIDB_TRX`](/information-schema/information-schema-tidb-trx.md)
+
+## TIDB_SHARD
+
+`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。当二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在因单调递增或递减而导致的热点时，索引的前缀 `tidb_shard(a)` 将热点打散，从而提升集群扩展性。
+
+在高吞吐写入、点查询、批量点查询场景下，你可以在保留原有的查询条件的情况下使用该函数，以提升写入性能。
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:337]:
```suggestion
在高吞吐写入、点查询、批量点查询场景中，你可以在保留原有的查询条件的情况下使用该函数，以提升写入性能。
```

File: functions-and-operators/tidb-functions.md (hunk 1 above)
Comment (TonsnakeLin) [reply functions-and-operators/tidb-functions.md:361]: I have added something about how to create a shard index by tidb_shard and the usage scenario. The detail for shard index I think it's not suitable to describe here.

**Review by cfzjywxk — APPROVED** [review]

**Comment by TomShawn** [comment]
> [CLA assistant check](https://cla-assistant.io/pingcap/docs-cn?pullRequest=8775) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/pingcap/docs-cn?pullRequest=8775) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pingcap/docs-cn?pullRequest=8775) it.

@TonsnakeLin Please sign the CLA. Thanks!

File: functions-and-operators/tidb-functions.md [hunk 7]
```diff
@@ -329,3 +329,56 @@ select tidb_decode_sql_digests(@digests, 10);
 
 - [`Statement Summary Tables`](/statement-summary-tables.md)
 - [`INFORMATION_SCHEMA.TIDB_TRX`](/information-schema/information-schema-tidb-trx.md)
+
+## TIDB_SHARD
+
+`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。当二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在因单调递增或递减而导致的热点时，索引的前缀 `tidb_shard(a)` 将热点打散，从而提升集群扩展性。
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:335]:
```suggestion
`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。当二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在因单调递增或递减而产生的热点时，索引的前缀 `tidb_shard(a)` 会打散热点，从而提升集群扩展性。
```

File: functions-and-operators/tidb-functions.md [hunk 8]
```diff
@@ -329,3 +329,56 @@ select tidb_decode_sql_digests(@digests, 10);
...
+
+在高吞吐写入、点查询、批量点查询场景下，你可以在保留原有的查询条件的情况下使用该函数，以提升写入性能。
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:337]:
```suggestion
`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。当二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在因单调递增或递减而产生的热点时，索引的前缀 `tidb_shard(a)` 会打散热点，从而提升集群扩展性。

在高吞吐写入、点查询、批量点查询场景中，你可以在保留原有的查询条件的情况下使用该函数，以提升写入性能。
```

Comment (TonsnakeLin) [reply functions-and-operators/tidb-functions.md:337]: remove  “在高吞吐写入、点查询、批量点查询场景中，你可以在保留原有的查询条件的情况下使用该函数，以提升写入性能”

File: functions-and-operators/tidb-functions.md [hunk 9]
```diff
@@ -329,3 +329,56 @@ select tidb_decode_sql_digests(@digests, 10);
...
+
+> **注意 SHARD INDEX 使用场景限制：**
+>
+> - 二级唯一索引上 key 值存在单调递增或递减导致的写入热点，且该索引包含的列是整形值
+> - 业务中 SQL 语句根据该二级索引的全部字段做等值查询，查询可以是单独的 SELECT，也可以是 UPDATE/DELETE 等产生的内部查询，等值查询包括 "a = 1" 或 "a IN (1, 2, ......)" 两种方式
+
+```SQL
+create table test(id int primary key clustered, a int, b int, unique key uk((tidb_shard(a)), a)); 
+```
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:346]: 挪到上方
```suggestion
```

File: functions-and-operators/tidb-functions.md [hunk 10]
```diff
@@ -329,3 +329,56 @@ select tidb_decode_sql_digests(@digests, 10);
...
+
+上述示例会使用 `TIDB_SHARD` 函数计算 `12373743746` 的 SHARD 值。
+
+{{< copyable "sql" >}}
+
+```sql
+create table test(id int primary key clustered, a int, b int, unique key uk((tidb_shard(a)), a)); 
+```
+
+上述示例使用 `TIDB_SHARD` 函数创建 SHARD INDEX。
```
Comment (en-jin19) [inline functions-and-operators/tidb-functions.md:380]:
```suggestion
- 使用 `TIDB_SHARD` 函数计算 【XXX】 的 SHARD 值

    以下示例提供如何使用 `TIDB_SHARD` 函数计算 `12373743746` 的 SHARD 值。

    {{< copyable "sql" >}}

    ```sql
    select TIDB_SHARD(12373743746);
    ```

    ```sql
    +-------------------------+
    | TIDB_SHARD(12373743746) |
    +-------------------------+
    |                     184 |
    +-------------------------+
    1 row in set (0.00 sec)
    ```

- 使用 `TIDB_SHARD` 函数创建 SHARD INDEX

    {{< copyable "sql" >}}

    ```sql
    create table test(id int primary key clustered, a int, b int, unique key uk((tidb_shard(a)), a)); 
    ```
```

File: functions-and-operators/tidb-functions.md [hunk 11]
```diff
@@ -329,3 +329,71 @@ select tidb_decode_sql_digests(@digests, 10);
 
 - [`Statement Summary Tables`](/statement-summary-tables.md)
 - [`INFORMATION_SCHEMA.TIDB_TRX`](/information-schema/information-schema-tidb-trx.md)
+
+## TIDB_SHARD
+
+`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。SHARD INDEX 是一种以`TIDB_SHARD` 函数为前缀的表达式索引。
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:335]:
```suggestion
`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。SHARD INDEX 是一种以 `TIDB_SHARD` 函数为前缀的表达式索引。
```

File: functions-and-operators/tidb-functions.md [hunk 12]
```diff
@@ -329,3 +329,71 @@ select tidb_decode_sql_digests(@digests, 10);
...
+
+### SHARD INDEX
+
+- 创建方式：
+
+    当二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在因单调递增或递减而产生的热点时，索引的前缀 `tidb_shard(a)` 会打散热点，从而提升集群扩展性。
+
+- 适用场景：
+
+    - 二级唯一索引上 key 值存在单调递增或递减导致的写入热点，且该索引包含的列是整形值
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:345]:
```suggestion
    - 二级唯一索引上 key 值存在单调递增或递减导致的写入热点，且该索引包含的列是整型。
```

File: functions-and-operators/tidb-functions.md [hunk 13]
```diff
@@ -329,3 +329,71 @@ select tidb_decode_sql_digests(@digests, 10);
...
+    - 业务中 SQL 语句根据该二级索引的全部字段做等值查询，查询可以是单独的 SELECT，也可以是 UPDATE/DELETE 等产生的内部查询，等值查询包括 "a = 1" 或 "a IN (1, 2, ......)" 两种方式
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:346]:
```suggestion
    - 业务中 SQL 语句根据该二级索引的全部字段做等值查询，查询可以是单独的 `SELECT`，也可以是 `UPDATE`，`DELETE` 等产生的内部查询，等值查询包括 `a = 1` 或 `a IN (1, 2, ......)` 两种方式。
```

File: functions-and-operators/tidb-functions.md [hunk 14]
```diff
@@ -329,3 +329,71 @@ select tidb_decode_sql_digests(@digests, 10);
...
+        - 非等值或 IN 查询无法使用索引
+        - 查询条件中 AND 和 OR 混合且最外层是 AND 算子时无法使用 Shard Index索引
+        - GROUP BY 无法使用 Shard Index
+        - ORDER BY 无法使用 Shard Index
+        - ON 子句无法使用 Shard Index
+        - WHERE 子查询无法使用 Shard Index
+        - Shard Index 只能打散整形字段的唯一索引
+        - Shard Index 联合索引可能失效
+        - Shard Index 无法走 FastPlan流程
+        - Shard Index 无法使用 Prepare Plan Cache
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:359]:
```suggestion
    - 非等值或 `IN` 查询无法使用索引。
    - 查询条件中 `AND` 和 `OR` 混合且最外层是 `AND` 算子时无法使用 SHARED INDEX。
    - `GROUP BY` 无法使用 SHARED INDEX。
    - `ORDER BY` 无法使用 SHARED INDEX。
    - `ON` 子句无法使用 SHARED INDEX。
    - `WHERE` 子查询无法使用 SHARED INDEX。
    - SHARED INDEX 只能打散整型字段的唯一索引。
    - SHARED INDEX 联合索引可能失效。
    - SHARED INDEX 无法走 FastPlan 流程。
    - SHARED INDEX 无法使用执行计划缓存。
```

File: functions-and-operators/tidb-functions.md [hunk 15]
```diff
@@ -329,3 +329,71 @@ select tidb_decode_sql_digests(@digests, 10);
...
+### 语法图
+
+```
+TIDBShardExpr ::=
+    "TIDB_SHARD" "(" expr ")"
+```
+
+### 示例
+
+- 使用 `TIDB_SHARD` 函数计算 【XXX】 的 SHARD 值
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:370]:
```suggestion
- 使用 `TIDB_SHARD` 函数计算 SHARD 值
```

File: functions-and-operators/tidb-functions.md [hunk 16]
```diff
@@ -329,3 +329,71 @@ select tidb_decode_sql_digests(@digests, 10);
...
+
+    以下示例提供如何使用 `TIDB_SHARD` 函数计算 `12373743746` 的 SHARD 值。
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    select TIDB_SHARD(12373743746);
+    ```
+
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:379]:
```suggestion

计算得出 SHARD 值为：

```

File: functions-and-operators/tidb-functions.md [hunk 17]
```diff
@@ -329,3 +329,71 @@ select tidb_decode_sql_digests(@digests, 10);
...
+    +-------------------------+
+    1 row in set (0.00 sec)
+    ```
+
+- 使用 `TIDB_SHARD` 函数创建 SHARD INDEX
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    create table test(id int primary key clustered, a int, b int, unique key uk((tidb_shard(a)), a));
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:394]:
```suggestion
    create table test(id int primary key clustered, a int, b int, unique key uk((tidb_shard(a)), a));
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

**Review by Oreoxmt — APPROVED** [review]

File: functions-and-operators/tidb-functions.md [hunk 18]
```diff
@@ -329,3 +329,73 @@ select tidb_decode_sql_digests(@digests, 10);
...
+    - `ON` 子句无法使用 SHARED INDEX。
+    - `WHERE` 子查询无法使用 SHARED INDEX。
+    - SHARED INDEX 只能打散整型字段的唯一索引。
+    - SHARED INDEX 联合索引可能失效。
+    - SHARED INDEX 无法走 FastPlan 流程。
+    - SHARED INDEX 无法使用执行计划缓存。
+
+### 语法图
+
+```
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:363]:
````suggestion
```ebnf+diagram
````

File: functions-and-operators/tidb-functions.md [hunk 19]
```diff
@@ -329,3 +329,73 @@ select tidb_decode_sql_digests(@digests, 10);
...
+TIDBShardExpr ::=
+    "TIDB_SHARD" "(" expr ")"
+```
+
+### 示例
+
+- 使用 `TIDB_SHARD` 函数计算 SHARD 值
+
+    以下示例提供如何使用 `TIDB_SHARD` 函数计算 `12373743746` 的 SHARD 值。
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:372]:
```suggestion
    以下示例说明如何使用 `TIDB_SHARD` 函数计算 `12373743746` 的 SHARD 值。
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: functions-and-operators/tidb-functions.md [hunk 20]
```diff
@@ -329,3 +329,73 @@ select tidb_decode_sql_digests(@digests, 10);
...
+
+## TIDB_SHARD
+
+`TIDB_SHARD` 函数用于创建一个 SHARD INDEX 来打散热点索引。SHARD INDEX 是一种以 `TIDB_SHARD` 函数为前缀的表达式索引。
+
+### SHARD INDEX
+
+- 创建方式：
+
+    当二级唯一索引 `uk((tidb_shard(a)), a))` 的索引字段 `a` 上存在因单调递增或递减而产生的热点时，索引的前缀 `tidb_shard(a)` 会打散热点，从而提升集群扩展性。
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:341]: 这句话说的像是触发场景？并没有说如何创建

Comment (Oreoxmt) [reply functions-and-operators/tidb-functions.md:341]: https://github.com/pingcap/docs-cn/pull/9670 已提 PR 添加创建方式的说明“使用 `uk((tidb_shard(a)), a))` 为字段 `a` 创建一个 SHARD INDEX。”

**Pull request merged**